cmake/modules/extensions.cmake

```diff
@@ -718,6 +718,222 @@ function(generate_inc_file_for_target
+#                   current date and time, set this option to "now".
+#
+# Examples:
+#
+#   generate_binary_embed_inc_file_for_target(app
+#     OUTPUT        ${gen_dir}/test_data_embed_combined.inc
+#     BINARY_FILES  test1.bin test2.bin
+#     SYMBOL_NAMES  symbol_names test1_data test2_data
+#     ALIGN         16
+#   )
+#
+#   generate_binary_embed_inc_file(
+#     OUTPUT        ${gen_dir}/test_data_embed.inc
+#     BINARY_FILES  test_data.bin
+#     SYMBOL_NAMES  test_data
+#     OFFSET        5
+#     LENGTH        52
+#     SECTION       tdata
+#     READONLY      no
+#     ALIGN         32
+#   )
+#
+#   generate_binary_embed_inc_file(
+#     OUTPUT        ${gen_dir}/raw_data.inc
+#     BINARY_FILES  ${EXTERNAL_SDK_PATH}/testx_file.bin
+#     SYMBOL_NAMES  test_data
+#   )
+#
+function(generate_binary_embed_inc_file_for_target target)
+  set(single_args "OUTPUT")
+  set(opt_single_args "SECTION;ALIGN;READONLY;OFFSET;LENGTH;COMPRESS;GZIP_MTIME")
+  set(multi_args "BINARY_FILES;SYMBOL_NAMES")
+  cmake_parse_arguments(PARSE_ARGV 1 BINARY_EMBED "${options}"
+    "${single_args};${opt_single_args}" "${multi_args}")
+
+  if(BINARY_EMBED_UNPARSED_ARGUMENTS)
+    message(FATAL_ERROR "generate_binary_embed_inc_file_for_target(${ARGV0} ...) given unknown "
+                        "arguments: ${BINARY_EMBED_UNPARSED_ARGUMENTS}"
+    )
+  endif()
+
+  foreach(arg ${single_args};${multi_args})
+    if(NOT DEFINED BINARY_EMBED_${arg})
+      message(FATAL_ERROR "generate_binary_embed_inc_file_for_target(${ARGV0} ...) "
+                          "missing required argument: ${arg}"
+      )
+    endif()
+  endforeach()
+
+  list(LENGTH BINARY_EMBED_BINARY_FILES bin_files_count)
+  list(LENGTH BINARY_EMBED_SYMBOL_NAMES symbol_names_count)
+  if(NOT symbol_names_count EQUAL bin_files_count)
+      message(FATAL_ERROR "Number of specified binary files (${bin_files_count}) is not the same "
+              "as number of the corresponding symbols names (${symbol_names_count})")
+  endif()
+
+  set(generated_file ${BINARY_EMBED_OUTPUT})
+  set(binary_files ${BINARY_EMBED_BINARY_FILES})
+  set(symbol_names ${BINARY_EMBED_SYMBOL_NAMES})
+  set(section ${BINARY_EMBED_SECTION})
+
+  if(NOT DEFINED BINARY_EMBED_READONLY)
+    set(readonly yes)
+  else()
+    set(readonly ${BINARY_EMBED_READONLY})
+  endif()
+
+  if(NOT DEFINED BINARY_EMBED_ALIGN)
+    set(align 4)
+  else()
+    set(align ${BINARY_EMBED_ALIGN})
+  endif()
+
+  if(NOT DEFINED BINARY_EMBED_OFFSET)
+    set(offset 0)
+  else()
+    set(offset ${BINARY_EMBED_OFFSET})
+  endif()
+
+  if(NOT DEFINED BINARY_EMBED_LENGTH)
+    set(length -1)
+  else()
+    set(length ${BINARY_EMBED_LENGTH})
+  endif()
+
+  if(NOT DEFINED BINARY_EMBED_COMPRESS)
+    set(compress no)
+  else()
+    set(compress ${BINARY_EMBED_COMPRESS})
+    if (compress STREQUAL gzip)
+      set(compress_param -g)
+    else()
+      message(WARNING "Unusupported compression method.")
```

Comment:
afaict then according to this line:
https://github.com/zephyrproject-rtos/zephyr/blob/e1726295de1905cdd2c9fc2e3112710d5fa449bd/cmake/modules/extensions.cmake#L763
then the call:
```
generate_binary_embed_inc_file_for_target(foo ... COMPRESS no)
```
is a valid call but will trigger this warning.